bundles/binding/org.openhab.binding.denon/src/main/java/org/openhab/binding/denon/internal/DenonConnector.java

```diff
@@ -0,0 +1,607 @@
+	 */
+	private String translateVolumeCommand(String command) {
+		if (command.matches("Z[0-9]")) {
+			command = command + DenonProperties.PROPERTY_ZONE_VOLUME;
+		}
+		
+		return command;
+	}
+	
+	private State getStateForValue(String value) {
+		if (StringUtils.isBlank(value))
+			return UnDefType.UNDEF;
+		
+		return new StringType(value);
+	}
+	
+	/**
+	 * Display info could contain some garbled text, attempt to clean it up.   
+	 */
+	private String cleanupDisplayInfo(String titleValue) {
+		byte firstByteRemoved[] =  Arrays.copyOfRange(titleValue.getBytes(), 1, titleValue.getBytes().length);
+		titleValue = new String(firstByteRemoved).replaceAll("[\u0000-\u001f]", "");
+		return titleValue;
+	}
+
+	private AsyncHttpClientConfig createAsyncHttpClientConfig() {
+		Builder builder = new AsyncHttpClientConfig.Builder();
+		builder.setRequestTimeoutInMs(REQUEST_TIMEOUT_MS);
+		builder.setUseRawUrl(true);
+		return builder.build();
+	}
+	
+	private <T> T getDocument(String uri, Class<T> type) {
+		try {
+			HttpURLConnection connection =
+				    (HttpURLConnection) new URL(uri).openConnection();
+			connection.setRequestMethod("GET");
+			connection.setConnectTimeout(REQUEST_TIMEOUT_MS);
+			connection.setReadTimeout(REQUEST_TIMEOUT_MS);
+			
+			JAXBContext jc = JAXBContext.newInstance(type);
+			InputStream xml = connection.getInputStream();
+			
+			XMLInputFactory xif = XMLInputFactory.newInstance();
+		    XMLStreamReader xsr = xif.createXMLStreamReader(xml);
+		    xsr = new PropertyRenamerDelegate(xsr);
+
+			@SuppressWarnings("unchecked")
+			T obj = (T) jc.createUnmarshaller().unmarshal(xsr);
+			connection.disconnect();
+			
+			return obj;
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (JAXBException e) {
+			e.printStackTrace();
+		} catch (XMLStreamException e) {
+			e.printStackTrace();
+		}
+		
+		return null;
+	}
+	
+	private <T,S> T postDocument(String uri, Class<T> type, S input) {
+		try {
+			JAXBContext jaxbContext = JAXBContext.newInstance(input.getClass());
+			Marshaller jaxbMarshaller = jaxbContext.createMarshaller();
+			 
+			HttpURLConnection connection =
+			    (HttpURLConnection) new URL(uri).openConnection();
+			connection.setRequestMethod("POST");
+			connection.setDoOutput(true);
+			connection.setConnectTimeout(REQUEST_TIMEOUT_MS);
+			connection.setReadTimeout(REQUEST_TIMEOUT_MS);
+			
+			jaxbMarshaller.marshal(input, connection.getOutputStream());
+			
+			JAXBContext jc = JAXBContext.newInstance(type);
+			InputStream xml = connection.getInputStream();
+			
+			@SuppressWarnings("unchecked")
+			T obj =
+			    (T) jc.createUnmarshaller().unmarshal(xml);
+			connection.disconnect();
+			
+			return obj;
+		} catch (JAXBException e) {
+			e.printStackTrace();
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		
+		return null;
+	}
```

Comment:
rather than using this bare implementations please rather use the HttpUtil which encapsulates most of the HTTP details for you